### PR TITLE
MYST-445 Make install and readme instructions great again

### PR DESCRIPTION
**Pull request by tadovas**
Key points:
1. Runtime and config dirs defaults changed to current dir of executable
2. Persistent data dir left untouched (home of user)
3. Updated install and readme instructions

### CODE REVIEW

File: INSTALL.md [hunk 1]
```diff
@@ -1,50 +1,26 @@
...
-sudo yum install docker-ce
-```
-
-##### Fetching and running docker image
-```bash
-sudo docker run --cap-add NET_ADMIN --net host --publish "1194:1194" --name mysterium-node -d mysteriumnetwork/mysterium-node:{VERSION} --agreed-terms-and-conditions
-```
-You can skip `{VERSION}` to fetch latest image.
-
 Note: to run server, you will have to accept terms & conditions by adding '--agreed-terms-and-conditions' command line option.
```
Comment (Waldz) [inline INSTALL.md]: Suggest to improve instruction like here: https://hub.docker.com/r/mysteriumnetwork/mysterium-node/

Comment (tadovas) [reply INSTALL.md]: Both updated.

File: INSTALL.md [hunk 2]
```diff
@@ -1,50 +1,26 @@
...
 https://hub.docker.com/r/mysteriumnetwork/mysterium-node/
-### Installation
-
-Read on how to setup docker repository for Debian [here](https://docs.docker.com/install/linux/docker-ce/debian/) 
-and for Ubuntu [here](https://docs.docker.com/install/linux/docker-ee/ubuntu/).
 
-##### Debian / Ubuntu
+### Fetching and running docker image
 ```bash
-sudo apt-get install docker-ce
```
[the remark this answers is not in the document]
Comment (tadovas) [reply INSTALL.md]: Fixed (should be)

File: INSTALL.md [hunk 3]
```diff
@@ -45,6 +21,7 @@ sudo docker logs -f mysterium-node
 
 
 ## Mysterium VPN node (Debian && Ubuntu) - tested on Ubuntu 16.04
+Note: you need to replace {version} with specific version number from [releases](https://github.com/MysteriumNetwork/node/releases/)
 ### Download
  * https://github.com/MysteriumNetwork/node/releases/download/{VERSION}/mysterium-node_linux_amd64.deb
  * https://github.com/MysteriumNetwork/node/releases/download/{VERSION}/mysterium-node_linux_armhf.deb
```
Comment (zolia) [inline INSTALL.md]: we no longer build for 'armhf'.

File: INSTALL.md [hunk 4]
```diff
@@ -121,32 +98,16 @@ sudo mysterium_client --data-dir=/var/lib/mysterium-client --config-dir=/etc/mys
...
-## Mysterium VPN client (standalone Linux binaries)
-### Download
- * https://github.com/MysteriumNetwork/node/releases/download/{VERSION}/mysterium_client_linux_amd64
- * https://github.com/MysteriumNetwork/node/releases/download/{VERSION}/mysterium_client_linux_armhf
 ### Running
 ```bash
-mysterium_client --help
-sudo mysterium_client && mysterium_client --cli
+cd <extracedir>
+sudo ./<mysterium_client or mysterium_server>
```
Comment (zolia) [inline INSTALL.md]: bit overkill, i'd just split into two commands.

File: cmd/directories.go [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+		return err
+	}
+
+	return ensureOrCreateDir(options.Data)
+}
+
+func ensureOrCreateDir(dir string) error {
+	err := ensureDirExists(dir)
+	if os.IsNotExist(err) {
+		return os.MkdirAll(dir, 0600)
```
Comment (zolia) [inline cmd/directories.go]: I would add some verbosity here.. "missing {dir} creating.. "

File: INSTALL.md (hunk 3 above)
Comment (tadovas) [reply INSTALL.md]: Removed

File: INSTALL.md (hunk 4 above)
Comment (tadovas) [reply INSTALL.md]: Splitted into separate sections for client and node

File: cmd/directories.go (hunk 5 above)
Comment (tadovas) [reply cmd/directories.go]: Added log line if attempting to create a dir

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -19,29 +19,25 @@ glide install
 go build github.com/mysterium/node
```
Comment (Waldz) [inline CONTRIBUTING.md]: This is not valid also

Comment (tadovas) [reply CONTRIBUTING.md]: Step changed from "Compile code" to "Fetch dependencies"

**Pull request merged**